File: DeskBuilder/src/main/java/com/ucg/desk/Card.java
package com.ucg.desk;

import android.net.Uri;

import com.ucg.desk.util.CardUtils;

/**
 * Created by sone on 25.02.14.
 */
public class Card {

    private String mType;
    private String mFileName;
    private String mPath;

    public Card(final String type, final String fileName) {
        init(type, fileName);
    }

    public String getType() {
        return mType;
    }

    public void setType(final String type) {
        mType = type;
        init(type, mFileName);
    }

    public String getFileName() {
        return mFileName;
    }

    public void setFileName(final String fileName) {
        mFileName = fileName;
        init(mType, fileName);
    }

    private void init(final String type, final String fileName){
        mType = type;
        mFileName = fileName;
        mPath = type + "/" + fileName;
    }

    public String getPath(){
        return mPath;
    }

    public Uri getUri() {
        return Uri.parse(CardUtils.FILE_ANDROID_ASSET + getPath());
    }
}

File: DeskBuilder/src/main/scala/com/ucg/desk/Card.scala
package com.ucg.desk

import android.net.Uri
import com.ucg.desk.util.CardUtils
import com.elegion.dk.sqlite.{SQLite, SQLiteColumn, SQLiteTableEditor, SQLiteTable}
import com.ucg.desk.sqlite.SQLiteProvider
import android.provider.BaseColumns
import android.database.sqlite.SQLiteDatabase

class Card(var kind: String, var fileName: String) extends SQLiteTable with CardSchema {

  def this() = this("", "")
  def this(path:String) = this(path.split("/")(0), path.split("/")(1))

  override def getTableName: String = TABLE

  override def onCreate(db: SQLiteDatabase) {
    new SQLiteTableEditor(this)
      .addColumn(new SQLiteColumn(Columns.KIND))
      .addColumn(new SQLiteColumn(Columns.FILENAME))
      .create(db)
  }

  def path = kind + "/" + fileName

  def getUri: Uri = {
    Uri.parse(CardUtils.FILE_ANDROID_ASSET + path)
  }

}

trait CardSchema {
  val TABLE = "card"
  val URI: Uri = SQLite.uri(SQLite.SCHEME, SQLiteProvider.AUTHORITY, TABLE)

  object Columns extends BaseColumns {
    val KIND = "kind"
    val FILENAME = "filename"
  }

}